integration/apps/geopmbench/geopmbench.py

```diff
@@ -88,3 +88,39 @@ def get_bash_exec_path(self):
 
     def get_bash_exec_args(self):
         return self._bench_conf.get_path()
+
+class GeopmbenchAppConf(apps.AppConf):
+    ''' A version that user can pass config files.'''
+    @staticmethod
+    def name():
+        return 'geopmbench'
+
+    def __init__(self, bench_conf_path, ranks_per_node):
+        self._bench_conf_path = bench_conf_path
+        self._ranks_per_node = ranks_per_node
+
+    def get_rank_per_node(self):
+        return self._ranks_per_node
+
+    def get_bash_exec_path(self):
+        # TODO: may need to find local version if not installed
+        return 'geopmbench'
+
+    def get_bash_exec_args(self):
+        return self._bench_conf_path
+
+def setup_geopmbench_run_args(parser):
+    """ Add common arguments for all run scripts.
+    """
+    parser.add_argument('config_file', action='store', type=str,
```

Comment:
I think `config_file` should be a required argument.

```
required=True
```

Otherwise get_bash_exec_args() will return None and the string "None" will be added to the bash command line.

Reply:
I believe ‘required’ is for optional arguments not for positional arguments: https://docs.python.org/3/library/argparse.html#required (edited) 

config_file does not start with ‘-’ or ‘--’ so it is a positional argument.

Reply:
You are correct.

Reply:
I added a comment to the function to make it clear.